Code/Framework/AzNetworking/AzNetworking/UdpTransport/UdpConnection.cpp

```diff
@@ -88,12 +88,14 @@ namespace AzNetworking
 
     bool UdpConnection::SendReliablePacket(const IPacket& packet)
     {
+        AZStd::lock_guard lock(m_sendPacketMutex);
```

Comment:
nit: Do we need to guard the both the sequence id and send steps. Maybe useful to cover what the lock is for here.

Looks like we're guarding both. Assume its because SendPacket is not multithread aware but curious as to why.

Reply:
Correct, SendPacket is not multi-threaded, so we need to mutex the whole call here. Sequence is is managed inside by UdpNetworkInterface.

Reply:
Are we scoped locking for each packet send? What's the long term feasibility of scoped locking once for a batch of packets?